packages/image_picker/image_picker/ios/Classes/FLTImagePickerPlugin.m

```diff
@@ -67,32 +119,30 @@ - (void)handleMethodCall:(FlutterMethodCall *)call result:(FlutterResult)result
   }
 
   if ([@"pickImage" isEqualToString:call.method]) {
-    _imagePickerController = [[UIImagePickerController alloc] init];
-    _imagePickerController.modalPresentationStyle = UIModalPresentationCurrentContext;
-    _imagePickerController.delegate = self;
-    _imagePickerController.mediaTypes = @[ (NSString *)kUTTypeImage ];
-
+    _phPickerFlag = NO;
     self.result = result;
     _arguments = call.arguments;
-
     int imageSource = [[_arguments objectForKey:@"source"] intValue];
 
-    switch (imageSource) {
-      case SOURCE_CAMERA: {
-        NSInteger cameraDevice = [[_arguments objectForKey:@"cameraDevice"] intValue];
-        _device = (cameraDevice == 1) ? UIImagePickerControllerCameraDeviceFront
-                                      : UIImagePickerControllerCameraDeviceRear;
-        [self checkCameraAuthorization];
-        break;
+    if (imageSource == SOURCE_GALLERY) {  // Capture is not possible with PHPicker
+      if (@available(iOS 14, *)) {
+        // PHPicker is used
+        _phPickerFlag = YES;
+        [self pickImageWithPHPicker:true];
+      } else {
+        // UIImagePicker is used
+        [self pickImageWithUIImagePicker];
       }
-      case SOURCE_GALLERY:
-        [self checkPhotoAuthorization];
-        break;
-      default:
-        result([FlutterError errorWithCode:@"invalid_source"
-                                   message:@"Invalid image source."
-                                   details:nil]);
-        break;
+    } else {
+      [self pickImageWithUIImagePicker];
+    }
+  } else if ([@"pickMultiImage" isEqualToString:call.method]) {
```

Comment:
Note: I think this is OK for now just to match the style of the old image picker APIs,

However, I think we should design a new API for easy configuration, such as
```dart
Future<List<PickedFile>?> pickMedia({
    double? maxWidth,
    double? maxHeight,
    int? quality,
    int limit = 1,
    List<MediaTypes>[] allowedMediaTypes = const <MediaTypes>[MediaTypes.image, MediaTypes.video],
  })
```

WDYT? This can be in future PRs tho.

Reply:
Yes, I think this could be a better approach. We can consider it in future PRs as you said.